File: GXpert/GXpert.Web/Modules/Users/Principal/PrincipalColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Users.Columns;

[ColumnsScript("Users.Principal")]
[BasedOnRow(typeof(PrincipalRow), CheckNames = true)]
public class PrincipalColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Email { get; set; }
    public string Mobile { get; set; }
    public string Username { get; set; }
    public string StateTitle { get; set; }
    public string DistrictTitle { get; set; }
    public string TalukaTitle { get; set; }
    public DateTime Dob { get; set; }
    public string InstituteName { get; set; }
  
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Users/PrincipalColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { PrincipalRow } from "./PrincipalRow";

export interface PrincipalColumns {
    Id: Column<PrincipalRow>;
    Name: Column<PrincipalRow>;
    Email: Column<PrincipalRow>;
    Mobile: Column<PrincipalRow>;
    Username: Column<PrincipalRow>;
    StateTitle: Column<PrincipalRow>;
    DistrictTitle: Column<PrincipalRow>;
    TalukaTitle: Column<PrincipalRow>;
    Dob: Column<PrincipalRow>;
    InstituteName: Column<PrincipalRow>;
}

export class PrincipalColumns extends ColumnsBase<PrincipalRow> {
    static readonly columnsKey = 'Users.Principal';
    static readonly Fields = fieldsProxy<PrincipalColumns>();
}